src/auth/client.ts

```diff
@@ -11,7 +11,7 @@ interface tokenInfo {
   peer: PeerId
 }
 
-export interface AuthenticateServerOptions extends AbortOptions {
+export interface AuthenticateServerOptions {
```

Comment:
What's the thinking behind making the `VerifyPeerOptions` a single-member interface?

Certainly the generated docs are easier to read if the options types accepted by methods are not unions, which is the approach taken elsewhere in the JS stack.

This also avoids a breaking change here due to the removal of `extends AbortOptions` which means any `signal` prop passed as a member of an object of this type would fail to compile.

```suggestion
export interface AuthenticateServerOptions extends RequestInit {
  /**
   * Accepts the PeerId of the server, return true if it was the expected
   * peer or false if not.
   */
  verifyPeer?(peerId: PeerId, options: AbortOptions): boolean | Promise<boolean>

```

This suggestion implies also removing the `VerifyPeerOptions` interface and removing the `AbortOptions` union from the `authenticateServer` method definition.

Reply:
No Opinion here. I'll defer to your opinion on what is more ergonomic to use.